pkg/config/config.go

```diff
@@ -188,6 +189,26 @@ func (c *Config) Validate(fs *flag.FlagSet) error {
 
 	c.Metrics.ServiceConfig.APIEnableGetConfiguration = c.EnableConfigEndpoints
 
+	// find every traces instance that is to be used by app agent receiver
+	// and configure it to include a PushReceiver to be able to push traces internally
+	// this cannot happen in unmarshall stage since integrations are not yet unmarshalled
+	if c.Integrations.configV2 != nil {
+		for _, conf := range c.Integrations.configV2.Configs {
+			if conf.Name() != app_agent_receiver.IntegrationName {
+				continue
+			}
+			appAgentReceiverConf := conf.(*app_agent_receiver.Config)
+			if appAgentReceiverConf.TracesInstance == "" {
+				continue
+			}
+			for idx, instanceConf := range c.Traces.Configs {
+				if instanceConf.Name == appAgentReceiverConf.TracesInstance {
+					c.Traces.Configs[idx].PushReceiver = true
+				}
+			}
+		}
+	}
```

Comment:
@mapno WDYT about having every tracing instance always have a PushReceiver? Domas mentioned you suggested only enabling it when needed, but I'm concerned about the effect it has on pkg/config here. 

Can you talk a little bit about the concerns you raised?

Reply:
Will do :)